openedx/adg/lms/certificates_override/helpers.py

```diff
@@ -0,0 +1,81 @@
+"""
+Helpers for certificates
+"""
+from django.utils import translation
+from django.utils.translation import gettext as _
+from opaque_keys.edx.keys import CourseKey
+
+from common.djangoapps.util.date_utils import strftime_localized
+from lms.djangoapps.certificates.models import CertificateStatuses, GeneratedCertificate
+from lms.djangoapps.certificates.views.webview import _get_user_certificate
+from lms.djangoapps.courseware.courses import get_course_by_id
+from openedx.adg.lms.certificates_override.constants import ARABIC_LANGUAGE_CODE
+from openedx.core.djangoapps.certificates.api import display_date_for_certificate
+
+
+def get_arabic_certificate_extra_context(request, course_id):
+    """
+    Compute `date_issued` and `gender` for the Arabic Course Certificates.
+
+    This method is used directly from arabic certificate template. This is done to avoid core changes and customise the
+    Arabic certificates. To display gender specific text, we need to send gender in context. Date formatting is also
+    custom for Arabic certificates so this method returns date in the required format as well.
+
+    Args:
+        request (WSGIRequest): request object
+        course_id (str): Course Id
+
+    Returns:
+        tuple: Gender of the learner and Date issued.
+    """
+    certificate = None
+    certificate_uuid = request.resolver_match.kwargs.get('certificate_uuid', None)
+    if certificate_uuid:
+        certificate = GeneratedCertificate.eligible_certificates.filter(
+            verify_uuid=certificate_uuid,
+            status=CertificateStatuses.downloadable
+        ).first()
+
+    user = certificate.user if certificate else request.user
+    course_key = CourseKey.from_string(course_id)
+    course = get_course_by_id(course_key)
+
+    if not certificate:
+        preview_mode = request.GET.get('preview', None)
+        certificate = _get_user_certificate(request, user, course_key, course, preview_mode)
+
+    date_format = u'{day} {month} {year}' if course.language == ARABIC_LANGUAGE_CODE else u'{month} {day}, {year}'
+    display_date = display_date_for_certificate(course, certificate)
+
+    with translation.override(ARABIC_LANGUAGE_CODE):
+        date_issued = _(date_format).format(  # pylint: disable=translation-of-non-string
+            month=strftime_localized(display_date, '%B'),
+            day=display_date.day,
+            year=display_date.year
+        )
+
+    return user.profile.gender, date_issued
+
+
+def set_certificate_recipient_font_size(accomplishment_copy_name):
+    """
+    Font size for user name on certificate.
+
+    Args:
+        accomplishment_copy_name (str): Name to be displayed on certificate.
+
+    Returns:
+        str: Font size.
+    """
+    size = len(accomplishment_copy_name)
+
+    if size > 32:
+        return '20px'
```

Comment:
Can we please define constants for pixel sizes?

Reply:
I don't think there is any need to create constants for this one, Also this is moved from mako to backend.